analyzers/hardcodedNonce.go

```diff
@@ -0,0 +1,231 @@
+		if valType.Parent() != nil {
+			trackedFunctions := make(map[string][]int)
+			for index, funcArgs := range valType.Parent().Params {
+				if funcArgs.Name() == valType.Name() && funcArgs.Type() == valType.Type() {
+					trackedFunctions[valType.Parent().String()] = []int{len(valType.Parent().Params), index}
+				}
+			}
+			result := iterateAndGetArgsFromTrackedFunctions(ssaFuncs, &trackedFunctions)
+
+			issueDescription += " by passing a parameter to a function and"
+			// recursively backtrack to where the origin of a variable passed to multiple functions is
+			for _, trackedVariable := range *result {
+
+				tmp, hasErr := raiseIssue(trackedVariable, &trackedFunctions, ssaFuncs, pass, issueDescription)
+				gosecIssue = append(gosecIssue, tmp...)
+				err = hasErr
+			}
+		}
+	}
+	return gosecIssue, err
+}
+
+// Iterate through all places that use the `variable` argument and check if it's used in one of the tracked functions
+func iterateThroughReferrers(variable *ssa.Value, funcsToTrack *map[string][]int, analyzerID string, issueDescription string, fileSet *token.FileSet, issueSeverity issue.Score, issueConfidence issue.Score) ([]*issue.Issue, error) {
+	if funcsToTrack == nil || variable == nil || analyzerID == "" || issueDescription == "" || fileSet == nil {
+		return nil, errors.New("received a nil object")
+	}
+	var gosecIssues []*issue.Issue = nil
+	// Go trough all functions that use the given arg variable
+	for _, referrer := range *(*variable).Referrers() {
+		// Iterate trough the functions we are interested
+		for trackedFunc := range *funcsToTrack {
+
+			// Split the functions we are interested in, by the '.' because we will use the function name to do the comparison
+			// MIGHT GIVE SOME FALSE POSITIVES THIS WAY
+			trackedFuncParts := strings.Split(trackedFunc, ".")
+			trackedFuncPartsName := trackedFuncParts[len(trackedFuncParts)-1]
+			if strings.Contains(referrer.String(), trackedFuncPartsName) {
+				gosecIssues = append(gosecIssues, newIssue(analyzerID, issueDescription, fileSet, referrer.Pos(), issueSeverity, issueConfidence))
+			}
+		}
+	}
+	return gosecIssues, nil
+}
+
+// Check whether a function contains a call to crypto/rand.Read in it's function body
+func isFuncContainsCryptoRand(funcCall *ssa.Function) (bool, error) {
+	if funcCall == nil {
+		return false, errors.New("passed ssa.Function object is nil")
+	}
+	for _, block := range funcCall.Blocks {
+		for _, instr := range block.Instrs {
+			if call, ok := instr.(*ssa.Call); ok {
+				if calledFunction, ok := call.Call.Value.(*ssa.Function); ok {
+					if calledFunction.Pkg != nil && calledFunction.Pkg.Pkg.Path() == "crypto/rand" && calledFunction.Name() == "Read" {
+						return true, nil
+					}
+				}
+			}
+		}
+	}
+	return false, nil
+}
+
+func addToVarsMap(value *ssa.Value, mapToAddTo *map[string]*ssa.Value) {
+	key := (*value).Name() + (*value).Type().String() + (*value).String() + (*value).Parent().String()
```

Comment:
You don't need to deference the `value` when calling all these methods. The compiler does that automatically, and it is also safer when the `value` pointer is nil.

Reply:
I've tried but the intellisense complains, I am not sure whether it's because I have to dereference it or just the intellisense is misleading:
![image](https://github.com/user-attachments/assets/7932aa5c-2cd4-4f30-be80-7ce4c35b0df4)

Sorry, but I am not sure, what you mean exactly?